Utilized the gspread package to populate a google spreadsheet with data

Humphery7 opened the pull request:
@elseagle 
@AmazingGrace-D 
@tobaadesugba 

created python script to get data from github user profile 
features -> (repository_name || language_used || description_of_repo || datetime repository was posted/last_updated) and 
Updated a Google SpreadSheet with this data using the Gspread Package

Humphery7 committed:
initial commit

elseagle commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+    worksheet = client.open(spreadsheet).worksheet(worksheet)
+    worksheet.clear() #clearning worksheet on each run of program
+    
+    def github_data(url = "https://github.com/Humphery7?tab=repositories"):
```
Why use a nested function here? Zen of python: "Simple is better than complex"

Humphery7 replied:
Ohh alright i actually didn't realize it made it more complex
it just cause i thought it made it look more cleaner.

but i will be looking out as to whether it is adding to the complexity now, i will change it

elseagle replied:
Aside from functionality, readability is very important in programming. This piece will definitely work without the nested approach here unless you're thinking OOP, which you can also shoot at!

elseagle commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+            Description += [desc.text for desc in driver.find_elements(by=By.CSS_SELECTOR, value='p[itemprop="description"]')]
+            Datetime_Posted += [time.text for time in driver.find_elements(by=By.CSS_SELECTOR,value='relative-time[class="no-wrap"]')] 
+        except Exception as e:
+            print("one or more of elements not found", e)
```
use logging library instead of print

Humphery7 replied:
alright, i will learn on the logging library now

elseagle commented on chromedriver.exe:
you can add your chrome drive to .gitignore

elseagle commented on LICENSE.chromedriver:
add to .gitignore

elseagle commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+    
+    client = gspread.service_account(filename=path)
+    worksheet = client.open(spreadsheet).worksheet(worksheet)
+    worksheet.clear() #clearning worksheet on each run of program
```
add comments above the line you intend to speak on

Humphery7 replied:
Sir i would like to ask a question, after effecting these changes, do i send it the code again and also please how??

elseagle replied:
Just git add, commit and push again

Humphery7 replied:
alright thanks sir

elseagle reviewed (changes requested):
Impressive start, kindly checkout the comments I dropped.

elseagle commented:
@Humphery7 the name of your branch shouldn't be auxiliary, but leave it for this project.

tobaadesugba commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
```
These metadata should be in the readme.md file

tobaadesugba commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+
+path = "documents/datasets_for_data_science/gspread_project/spartan-figure-402211-2241d0349aab.json"
```
I'd advice you use relative path (from your current working directory) instead of absolute path. This will avoid errors when running this code on another computer. Check the link below for more info on relative and absolute paths.

https://courses.cs.washington.edu/courses/cse140/13wi/file-interaction.html#:~:text=Absolute%20and%20Relative%20file%20paths&text=An%20absolute%20file%20path%20describes,For%20example%2C%20example_directory.

tobaadesugba commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+
+path = "documents/datasets_for_data_science/gspread_project/spartan-figure-402211-2241d0349aab.json"
```
You should also use a placeholder file here (like 'project_info.json') along with a disclaimer comment mentioning that's a placeholder and mentioning why it's there

Humphery7 replied:
thank you sir, although i don't fully understand this part, do you mean i should remove the "spartan-figure-402211-2241d0349aab.json" and replace with "project_info.json"  like giving it a generic name and explaining what it is??

tobaadesugba replied:
Yes, that is what I'm talking about. Then you add a comment above it or in front of it to mention that it is a placeholder filename and mention why you added it.

tobaadesugba commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,60 @@
+
+path = "documents/datasets_for_data_science/gspread_project/spartan-figure-402211-2241d0349aab.json"
+
+def update_spreadsheet(spreadsheet = "Gspread SpreadSheet",worksheet="Sheet1"):
```
Docstrings will be a great addition here. Use it to describe what your function is doing.

tobaadesugba commented on project-1/requirements.txt:
```diff
@@ -0,0 +1,5 @@
+#Selenium version 4.14.0
```
This is not the proper way to format a requirements.txt file.
Run `pip freeze > requirements.txt` to generate a requirements file. It would add all packages in the environment and not just the ones you used in your project, it's just so you can see the proper way to format the file.

tobaadesugba reviewed (changes requested):
Great first commit, check through my review for more details. Don't forget to water your code with helpful comments so you and any other person can understand what's going on easily.

Humphery7 committed:
fixed some issues realized through pull_request reviews

Humphery7 reviewed (commented):
@elseagle 
@tobaadesugba 
@AmazingGrace-D 

-> Removed nested function which added to complexity of program, changed to only one function update_spreadsheet()
-> Updated requirement.txt file
-> Added docstring for function
-> changed absolute path to a relative file path with the help of the os module
-> added License.chromedriver and chromedriver.exe to .gitignore file
-> updated readme to include input data, process, and output data 
-> added more comments to impove clarity and made comment on top of each line of code it is referring too 
-> changed print to logging library as it is more robust

Humphery7 committed:
added gspread_code.py which was left out in last commit

Humphery7 reviewed (commented):
@elseagle 
@tobaadesugba 
@AmazingGrace-D 

commit 3 contains gspread_code.py which was mistakenly left out in second commit made

tobaadesugba commented on project-1/requirements.txt:
```diff
@@ -0,0 +1,4 @@
+selenium==4.14.0
+gspread==5.11.3
+pandas @ file:///C:/ci/pandas_1650373458095/work
```
The new format you used is now proper, however, `file:///C:/ci/pandas_1650373458095/work` is only available on your computer. Use the version number instead or a link that's publicly available to anyone willing to replicate your project.

tobaadesugba reviewed (changes requested):
Great work applying the recommendations. I love that your comments explain your code well. Fix the requirements issue I mentioned, and you're good to go on my end.

Humphery7 committed:
Delete chromedriver.exe

chromedriver.exe file was deleted, as it has been already added to the .gitignore file

Humphery7 committed:
Delete LICENSE.chromedriver

LICENSE.chromedriver file was deleted as it has been already added to the .gitignore file

elseagle commented on project-1/gspread_code.py:
```diff
@@ -0,0 +1,79 @@
+        data =  pd.DataFrame(github_repo).reset_index(drop=True)
+    except Exception as e:
+        logging.critical("one or more elements were not found", e)
+        exit(0)
```
In place of exit, raise an error, you're also exiting with "0" which implies, the code all was good, are you sure that's the message you want to pass?

Humphery7 replied:
ohh alright thank you sir, the 0 was just a habit i learnt from youtube a long time, never knew that was what it meant till now, i will change it now

elseagle reviewed (commented):
Looking solid, just some comments left for you

Humphery7 committed:
corrected issues in gspread_code.py and requirements.txt, gave more info in README.md file

Humphery7 reviewed (commented):
@elseagle 
@tobaadesugba 
@AmazingGrace-D 

-> corrected issues raised in requirments.txt file
->changed exit(0) in code and instead raised a valueerror
->added more info to readme file
